pkg/gatewayserver/gatewayserver.go

```diff
@@ -654,38 +654,51 @@ func (gs *GatewayServer) handleUpstream(conn connectionEntry) {
 	}
 }
 
-// UpdateConnectionStats updates the stats for a single gateway connection.
-func (gs *GatewayServer) UpdateConnectionStats(ctx context.Context, conn *io.Connection) error {
-	return gs.statsRegistry.Set(ctx, conn.Gateway().GatewayIdentifiers, conn.Stats())
+var allTraffic = io.Traffic{Up: true, Down: true, Status: true}
+
+// UpdateAllConnectionStats updates the stats for a single gateway connection.
+func (gs *GatewayServer) UpdateAllConnectionStats(ctx context.Context, conn *io.Connection) error {
+	return gs.statsRegistry.Set(ctx, conn.Gateway().GatewayIdentifiers, conn.Stats(), allTraffic)
 }
 
 func (gs *GatewayServer) updateConnStats(conn connectionEntry) {
 	ctx := conn.Context()
 	logger := log.FromContext(ctx)
 
+	if gs.statsRegistry == nil {
```

Comment:
This function will (or should not) be called in this case

Reply:
This is just to make sure that we we don't continue with a `nil` registry for whatever reason, can remove

Reply:
That’s an implementation error for which we can panic

Reply:
Remove check; let the stack panic on nil pointer reference. It's really an implementation error and we cannot begin to add these kinds of assertions everywhere.